carma_messenger_vehicle_plugin/src/msger_veh_cfg.py

```diff
@@ -0,0 +1,171 @@
+ #   Copyright (C) 2022 LEIDOS.
+ #
+ #   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ #   use this file except in compliance with the License. You may obtain a copy of
+ #   the License at
+ #
+ #   http://www.apache.org/licenses/LICENSE-2.0
+ #
+ #   Unless required by applicable law or agreed to in writing, software
+ #   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ #   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ #   License for the specific language governing permissions and limitations under
+ #   the License.
+
+import json
+import logging
+
+class MsgerVehicleCfg:
+
+    def _setup_logging(self, level):
+        """
+        Set up logging with the specified level.
+
+        Parameters:
+        - level: The log level to set up.
+        type: str
+        """
+        numeric_level = getattr(logging, level.upper(), None)
+        if not isinstance(numeric_level, int):
+            raise ValueError('Invalid log level: %s' % level)
+        logging.basicConfig(level=numeric_level, format='%(asctime)s - %(levelname)s - %(message)s')
+
+    def __init__(self, cfg_path, log_level='INFO'):
+        """
+        Initializes the vehicle configuration manager by loading vehicle data from a JSON file.
+
+        Parameters:
+        - cfg_path: The file path to the JSON configuration file containing messenger vehicle data.
+        type: str
+        """
+
+        self._setup_logging(log_level)
+        self._msger_veh_dict = {}
+        self._veh_state_dict = {}
+        try:
+            with open(cfg_path, 'r') as cfg_file:
+                config_json = json.load(cfg_file)
+                for veh in config_json["msgerVehs"]:
+                    veh_id = veh["id"]
+                    if veh_id in self._msger_veh_dict:
+                        logging.warning(f"Duplicate vehicle ID detected and skipped: {veh_id}")
+                        continue
+                    self._msger_veh_dict[veh_id] = {
```

Comment:
Would be good to document the expected format of this dict somewhere as a comment.